reark/src/main/java/io/reark/reark/data/stores/cores/ContentProviderStoreCoreBase.java

```diff
@@ -110,34 +118,44 @@ protected ContentProviderStoreCoreBase(@NonNull final ContentResolver contentRes
     private void initialize() {
         contentResolver.registerContentObserver(getContentUri(), true, getContentObserver());
 
-        // Observable transforming updates and inserts to ContentProviderOperations
-        Observable<ContentProviderOperation> operationObservable = updateSubject
+        // Observable transforming inserts, updates and deletes to ContentProviderOperations
+        Observable<CoreOperation> operationObservable = operationSubject
                 .onBackpressureBuffer()
                 .observeOn(Schedulers.io())
-                .flatMap(pair -> createOperation(pair.first, pair.second));
+                .flatMap(this::createCoreOperation);
 
         // Group the operations to a list that should be executed in one batch. The default
         // grouping logic is suitable for pojo stores, but some stores may need to provide
-        // their own grouping logic, if for example buffering delays are undesirable.
+        // their own grouping logic if for example buffering delays are undesirable.
         updateSubscription = groupOperations(operationObservable)
                 .observeOn(Schedulers.computation())
-                .map(ArrayList::new)
-                .doOnNext(list -> Log.v(TAG, "Grouped list of " + list.size()))
-                .doOnNext(operations -> {
-                    try {
-                        ContentProviderResult[] result = contentResolver.applyBatch(getAuthority(), operations);
-                        Log.v(TAG, String.format("Applied %s operations", result.length));
-                    } catch (RemoteException | OperationApplicationException e) {
-                        Log.e(TAG, "Error applying operations", e);
-                    }
-                })
-                .flatMap(Observable::from)
-                .map(ContentProviderOperation::getUri)
-                .subscribe(locker::release,
+                .doOnNext(operations -> Log.v(TAG, "Grouped list of " + operations.size()))
+                .flatMap(this::applyOperations)
```

Comment:
I know this is part of an earlier implementation, but should this be a concat map? Now the execution order of the batches might change, which I suppose might be undesirable?